src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.bookshop.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderItemResponseDto;
+import mate.academy.bookshop.dto.order.OrderRequestDto;
+import mate.academy.bookshop.dto.order.OrderResponseDto;
+import mate.academy.bookshop.dto.status.StatusDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.Status;
+import mate.academy.bookshop.repository.OrderRepository;
+import mate.academy.bookshop.repository.OrderStatusRepository;
+import mate.academy.bookshop.service.OrderService;
+import mate.academy.bookshop.service.ShoppingCartService;
+import mate.academy.bookshop.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final Status.StatusName DEFAULT_STATUS_NAME
+            = Status.StatusName.PENDING;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final OrderStatusRepository statusRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto addOrder(Long userId, OrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartService.getNonEmptyCartByUserId(userId);
+        Order newOrder = createOrder(userId, shoppingCart, requestDto);
+        orderRepository.save(newOrder);
+        shoppingCartService.clearShoppingCart(shoppingCart.getId());
+        return orderMapper.toDto(newOrder);

```

Comment:
let's ensure that user who is making a request can only create an order only for himself and nobody else